core/trino-main/src/main/java/io/trino/metadata/Metadata.java

```diff
@@ -716,4 +717,9 @@ default boolean isMaterializedView(Session session, QualifiedObjectName viewName
      * Returns a table handle for the specified table name with a specified version
      */
     Optional<TableHandle> getTableHandle(Session session, QualifiedObjectName tableName, Optional<TableVersion> startVersion, Optional<TableVersion> endVersion);
+
+    /**
+        Returns maximal number of tasks that should be in writing to specific connector.
+     */
+    OptionalInt getMaxWriterTasks(Session session, String catalogName);
```

Comment:
Why do we need this from Connector?

Reply:
We need more flexibility in the context of different connectors. Also, it was a @hashhar 's request in the previous PR.